File: 0066-plus-one/0066-plus-one.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int n=digits.size()-1;
        while(digits[n]==9 && n>0){
            digits[n]=0;
            n--;
        }
        if(n==0 && digits[0]==9){
            digits[0]=1;
            digits.push_back(0);
            return digits;
        }
        digits[n]++;
        return digits;
    }
};

File: 0066-plus-one/0066-plus-one.cs
public class Solution {
    public int[] PlusOne(int[] digits) {
        List<int> ans=new();
        // start from least significant
        int i=digits.Length-1;
        bool easy=false;
        while(i>-1){
            if(digits[i]<9){
                int x=i+1;
                while(x<digits.Length){
                    digits[x]=0;
                    x++;
                }
                easy=true;
                digits[i]++;
                break;
            }
            i--;
        }
        if(easy)
            return digits;
        ans.Add(1);
        foreach(var n in digits)
            ans.Add(0);
        return ans.ToArray();
    }
}